honeypots documentation updated (#6559)
opencv/cvat: site/content/en/docs/manual/advanced/annotation-quality.md

---
title: 'Annotation quality & Honeypot'
linkTitle: 'Annotation quality'
weight: 14
description: 'How to check the quality of annotation in CVAT'
---

In CVAT, it's possible to evaluate the quality of annotation through
the creation of a **Ground truth** job, referred to as a Honeypot.

> CVAT compares all other jobs in the specified tasks against the

Reviewer: ```suggestion
To estimate the task quality, CVAT compares all other jobs in the task against the
```

established **Ground truth** job and calculates annotation quality
based on this comparison.

> > **Note** that Honeypot does not work with cubooids.

Reviewer: ```suggestion
> **Note** that quality estimation only supports 2d tasks. It supports all the annotation types except 2d cuboids. 

> Tracks are considered separate shapes and compared on the per-frame basis with other tracks and shapes.
```

See:

- [Ground truth job](#ground-truth-job)
- [Managing Ground Truth jobs: Import, Export, and Deletion](#managing-ground-truth-jobs-import-export-and-deletion)
  - [Import](#import)
  - [Export](#export)
  - [Delete](#delete)
- [Assessing data quality with Ground truth jobs](#assessing-data-quality-with-ground-truth-jobs)
  - [Quality data](#quality-data)
  - [Annotation quality settings](#annotation-quality-settings)
  - [GT conflicts in the CVAT interface](#gt-conflicts-in-the-cvat-interface)
- [Annotation quality \& Honeypot video tutorial](#annotation-quality--honeypot-video-tutorial)

## Ground truth job

> You don’t need to annotate the whole dataset twice, the annotation quality of a small part of
> the data will show the quality of annotation for the whole dataset.
> 
> You need to select several frames from the whole
> dataset depending on the size of the task.
> 
> For the quality assurance to function properly, each job should contain at
> least a few overlapping frames. For example, for **task that contains only
> 30 frames** to obtain adequate data, it's advisable to select **8-10 frames**.
> 
> Depending on the dataset size, **5-15% of the data is good enough** for quality estimation.

Reviewer: ```suggestion
A Ground Truth job is a way to tell CVAT where to store and get the "correct" 
annotations for task quality estimation. To estimate task quality, you need to 
create a Ground Truth job in the task, and annotate it. You don’t need to 
annotate the whole dataset twice, the annotation quality of a small part of
the data shows the quality of annotation for the whole dataset.

For the quality assurance to function correctly, the Ground Truth job must have
a small portion of the task frames, and the frames must be chosen randomly.
Depending on the dataset size and task complexity, **5-15% of the data is typically good enough** for quality estimation, while keeps extra annotation 
overhead acceptable. 

For example, in a typical task with 2000 frames, selecting just 5%, which is 100 frames extra frames to annotate, can be enough to estimate the annotation quality. If the task  contains only 30 frames, it's advisable to select **8-10 frames**, which is about 30%. It is more than 15%, but in the case of smaller datasets, we need more samples to estimate quality reliably.

```

To create a **Ground truth** job, do the following:

1. Create a [task](/docs/manual/basics/create_an_annotation_task/), and open the task page.
2. Click **+**.

   ![Create job](/images/honeypot01.jpg)

3. In the **Add new job** window, fill in the following fields:

   ![Add new job](/images/honeypot02.jpg)

   - **Job type**: Use the default parameter **Ground truth**.
   - **Frame selection method**: Use the default parameter **Random**.
   - **Quantity %**: Set the desired percentage of frames for the "ground truth" job. <br>Note, that upon using **Quantity %**, the **Frames** field will be automatically populated.
   - **Frame count**: Set the desired number of frames for the "ground truth" job. <br>Note, that upon using **Frames**, the **Quantity %** field will be automatically populated.

>    - **Seed**: (Optional) TBD

Reviewer: ```suggestion
   - **Seed**: (Optional) If you need to make the random selection reproducible, specify this number. It can be any integer number, the same value will yield the same random selection (given that the frame number is unchanged).
```

> 

Reviewer: ```suggestion
  > **Note** if you want to use a custom frame sequence, you can do this using the server API instead. Read more [here](https://opencv.github.io/cvat/docs/api_sdk/sdk/reference/apis/jobs-api/#create).
```

4. Click **Submit**.
5. Annotate frames, save your work.
6. Change the status of the job to **Completed**.
7. Change **Stage** to **Accepted**.

The **Ground truth** job will appear in the jobs list.

![Add new job](/images/honeypot03.jpg)

## Managing Ground Truth jobs: Import, Export, and Deletion

Annotations from **Ground truth** jobs are not included in the dataset export,
they also cannot be imported during task annotations import
or with automatic annotation for the task.

> Import, export, and delete options are available from

Reviewer: ```suggestion
Import, export, and delete options are available from the
```

job's menu.

![Add new job](/images/honeypot04.jpg)

### Import

If you want to import the **Ground truth** job, do the following.

1. Open the task, and find the **Ground truth** job in the jobs list.
2. Click on three dots to open the menu.
3. From the menu, select **Import job**.
4. Select import format, and select file.
5. Click **OK**.

> 

Reviewer: ```suggestion

Note that if there are imported annotations for the frames that exist in the task, but not included in the Ground Truth job, thy will be silently ignored. This way, you don't need to worry about "cleaning up" your Ground Truth annotations for the whole dataset before importing them. Importing annotations for the frames
that are not known in the task still raise errors.

```

### Export

To export the **Ground truth** job, do the following.

1. Open the task, and find a job in the jobs list.
2. Click on three dots to open the menu.
3. From the menu, select **Export job**.

### Delete

To delete the **Ground truth** job, do the following.

1. Open the task, and find the **Ground truth** job in the jobs list.
2. Click on three dots to open the menu.
3. From the menu, select **Delete**.

## Assessing data quality with Ground truth jobs

Once you've established the **Ground truth** job, proceed to annotate the dataset.

> CVAT will begin the quality comparison between the annotated task and the **Ground truth** job.

Reviewer: ```suggestion
CVAT will begin the quality comparison between the annotated task and the **Ground truth** job in this task once it is finished (on the `acceptance` stage and in the `completed` state).
```

> > Note, that the process of quality calculation may take up to several hours, depending on
> > the amount of data and labeled objects.

Reviewer: ```suggestion
> Note, that the process of quality calculation may take up to several hours, depending on
> the amount of data and labeled objects, and is **not updated immediately** after task updates.
```

| GT Conflicts            | Conflicts identified during quality assessment, including extra or missing annotations. Mouse over the **?** icon for a detailed conflict report on your dataset.                                 |
| Issues                  | Number of [opened issues](/docs/manual/advanced/review/). If no issues were reported, will show 0.                                                                                                |
| Quality report          | Quality report in JSON format.                                                                                                                                                                    |
| Ground truth job data   | "Information about ground truth job, including date, time, and number of issues.                                                                                                                  |
| List of jobs            | List of all the jobs in the task                                                                                                                                                                  |

<!--lint enable maximum-line-length-->

### Annotation quality settings

> 

Reviewer: ```suggestion

If you need to tweak some aspects of comparisons, you can do this via 
the Quality Settings menu. Here you can, for instance, configure, which overlap
should be considered low, or how annotations should be compared.

```

| Check object visibility                                                            | Check for partially-covered annotations. Masks and polygons will be compared to each other.                                                                                                                      |
| Min visibility threshold                                                           | Minimal visible area percent of the spatial annotations (polygons, masks)                                                                                      |
| for reporting covered annotations, useful with the Check object visibility option. |
| Match only visible parts                                                           | Use only the visible part of the masks and polygons in comparisons.                                                                                            |

<!--lint enable maximum-line-length-->

### GT conflicts in the CVAT interface

> 

Reviewer: ```suggestion

Once you know the quality summary, you may want to investigate 
the problems found in more detail. You can actually check and take action 
on the quality problems in jobs using the job "Review" mode. 

```

---
title: 'Annotation quality & Honeypot'
linkTitle: 'Annotation quality'
weight: 14
description: 'How to check the quality of annotation in CVAT'
---

In CVAT, it's possible to evaluate the quality of annotation through
the creation of a **Ground truth** job, referred to as a Honeypot.
To estimate the task quality, CVAT compares all other jobs in the task against the

> established **Ground truth** job and calculates annotation quality

Reviewer: ```suggestion
established **Ground truth** job, and calculates annotation quality
```

- [Assessing data quality with Ground truth jobs](#assessing-data-quality-with-ground-truth-jobs)
  - [Quality data](#quality-data)
  - [Annotation quality settings](#annotation-quality-settings)
  - [GT conflicts in the CVAT interface](#gt-conflicts-in-the-cvat-interface)
- [Annotation quality \& Honeypot video tutorial](#annotation-quality--honeypot-video-tutorial)

## Ground truth job

A **Ground truth** job is a way to tell CVAT where to store
and get the "correct" annotations for task quality estimation.

To estimate task quality, you need to
create a **Ground truth** job in the task,
and annotate it. You don’t need to
annotate the whole dataset twice,
the annotation quality of a small part of
the data shows the quality of annotation for
the whole dataset.

For the quality assurance to function correctly, the **Ground truth** job must
have a small portion of the task frames and the frames must be chosen randomly.
Depending on the dataset size and task complexity,
**5-15% of the data is typically good enough** for quality estimation,
while keeping extra annotation overhead acceptable.

For example, in a typical **task with 2000 frames**, selecting **just 5%**,

> which is 100 frames extra frames to annotate, **is enough** to estimate the

Reviewer: ```suggestion
which is 100 extra frames to annotate, **is enough** to estimate the
```

     <br>**Note** that when you use **Frames**, the **Quantity %** field will be will be autofilled.
   - **Seed**: (Optional) If you need to make the random selection reproducible, specify this number.
     It can be any integer number, the same value will yield the same random selection (given that the
     frame number is unchanged). <br> **Note** that if you want to use a
     custom frame sequence, you can do this using the server API instead,
     see [Jobs API #create](https://opencv.github.io/cvat/docs/api_sdk/sdk/reference/apis/jobs-api/#create).

4. Click **Submit**.
5. Annotate frames, save your work.
6. Change the status of the job to **Completed**.
7. Change **Stage** to **Accepted**.

The **Ground truth** job will appear in the jobs list.

![Add new job](/images/honeypot03.jpg)

## Managing Ground Truth jobs: Import, Export, and Deletion

Annotations from **Ground truth** jobs are not included in the dataset export,
they also cannot be imported during task annotations import
or with automatic annotation for the task.

Import, export, and delete options are available from the
job's menu.

![Add new job](/images/honeypot04.jpg)

### Import

> If you want to import the **Ground truth** job, do the following.

Reviewer: ```suggestion
If you want to import annotations into the **Ground truth** job, do the following.
```

1. Open the task, and find the **Ground truth** job in the jobs list.
2. Click on three dots to open the menu.
3. From the menu, select **Import job**.
4. Select import format, and select file.
5. Click **OK**.

> **Note** that if there are imported annotations for the frames that exist in the task,
> but are not included in the **Ground truth** job, they will be ignored.
> This way, you don't need to worry about "cleaning up" your **Ground truth**
> annotations for the whole dataset before importing them.
> Importing annotations for the frames that are not known in the task still raises errors.

### Export

> To export the **Ground truth** job, do the following.

Reviewer: ```suggestion
To export annotations from the **Ground truth** job, do the following.
```

1. Open the task, and find a job in the jobs list.
2. Click on three dots to open the menu.

> 3. From the menu, select **Export job**.

Reviewer: ```suggestion
3. From the menu, select **Export annotations**.
```
